### PR TITLE
piping urls from the server to the front end

### PR DESCRIPTION
**Pull request by xgess**
* a bunch of boilerplate for fetching some data (external_urls) and verifying it against a hash in the merkle root
* a local RPC for getting stellar exchange urls out of ^ and to the front end

depends on https://github.com/keybase/keybase/pull/3868

### CODE REVIEW

File: protocol/avdl/stellar1/local.avdl [hunk 1]
```diff
@@ -728,9 +728,22 @@ protocol local {
...
+    string title;
+    string description;
+    @jsonkey("icon_filename")
+    string iconFilename;
+    @jsonkey("admin_only")
+    boolean adminOnly;
+    string extra; // extra json tbd
+  }
+
+  array<ExchangeUrl> getExchangeUrlsLocal(int sessionID);
```
Comment (patrickxb) [inline protocol/avdl/stellar1/local.avdl]: I think these should be called ExternalURLs...or PartnerURLs.

So they aren't confused with the links we are adding to assets to the stellarx exchange.

**Comment by xgess** [comment]
> Anything frontend calls needs a sessionID.

lol. i figured something like that would be the case :/
ok thanks.

File: protocol/avdl/stellar1/local.avdl (hunk 1 above)
Comment (xgess) [reply protocol/avdl/stellar1/local.avdl]: done

File: go/libkb/constants.go [hunk 2]
```diff
@@ -723,6 +723,6 @@ const (
 const ProfileProofSuggestions = true
 
 const (
-	ExternalURLsBaseKey          = "external_urls"
-	ExternalURLsStellarExchanges = "stellar_exchanges"
+	ExternalURLsBaseKey         = "external_urls"
+	ExternalURLsStellarPartners = "stellar_exchanges"
```
Comment (patrickxb) [inline go/libkb/constants.go]: sorry to be a pain, but can you change the value of the const to `stellar_partners` too?

just don't want to get confused later...

thanks!

Comment (xgess) [reply go/libkb/constants.go]: done. https://github.com/keybase/keybase/pull/3891

**Pull request merged**